src/misc/generate-docs.ts

```diff
@@ -0,0 +1,102 @@
+import * as fs from 'fs'
+import * as os from 'os'
+import * as path from 'path'
+import * as yaml from 'js-yaml'
+
+//
+// SUMMARY
+//
+// This script rebuilds the usage section in the README.md to be consistent with the action.yml
+
+function updateUsage(
```

Comment:
we may want to move this function into a toolkit module

i added this based on feedback that the README.md should contain the inputs.

I didnt want to have to deal with maintaining both places, so now the CI runs the `npm run gendocs` and fails if it isnt up to date